libs/datasets/timeseries.py

```diff
@@ -292,6 +292,7 @@ def _timeseries_long(self) -> pd.Series:
             .sort_index()
         )
 
+    @lru_cache(maxsize=None)
```

Comment:
I guess it might not be a problem because the underlying dataset is final, but if the underlying dataframe changes it could cause inconsistencies is one thing - just a pattern that I think is worth using very sparingly if there aren't other options available

Reply:
100% can only do this when data is immutable. Part of the beauty of immutable objects is greater ease managing data derived from it.